MIGRATION.md

```diff
@@ -1,6 +1,21 @@
 # Migration Instructions for Major Releases
 
-### From `7.0.0` to `8.0.0`
+## Migrating to Trunk
```

Comment:
I'm wondering if we shouldn't already set this title to ``## Migrating from `8.x` to `9.x` ``, because, unlike the CHANGELOG for which we don't know what will be the next version number until we decide to do the release (depending on what PRs will have landed when we do the release and if there are breaking changes or minor changes or only bugfixes), for the Migration guide we know that it is specifically made to document what to change in client repos to account for **breaking changes**. So this section is guaranteed to be for a `9.x` release, not a `8.2.0` minor release or similar.

I'm mentioning this because, while for the `CHANGELOG.md` we have some automation that makes the section title be automatically updated by our `rake new_release` task to replace the mention of "Trunk" to the actual version number we're releasing and all… I don't think we've yet ported a similar logic for the `MIGRATION.md` file. Which means it would be really easy to do a new major release without remembering to manually update this section title to `9.0.0` at the time we release said `9.0.0` major version.

Maybe at some point we'll automate that, or maybe even add `Danger` to this repo to detect if the new CHANGELOG entry one added as part of a PR was under "Breaking Changes" and if so request that the `MIGRATION.md` file to be updated…? But in the meantime, I feel it'd be easier and ok to just use the already foreseen next major version in the title here.